### PR TITLE
Package-based dynamic add-on deployments

### PR DESCRIPTION
**Pull request by dremin**
### Summary

POC of an alternative approach for dynamically deploying services, using a package-based approach rather than Terraform.

Advantages:
- Much simpler
- Not limited to Twilio serverless projects--can be used for any npm package that needs its deployment orchestrated with the template
- When working with serverless projects, provides a consistent developer experience working with the serverless toolkit to test services locally, as well as taking advantage of existing deployment logic within the toolkit

Disadvantages:
- Does not provide an example of Terraform serverless deployments
- No mechanism for resolving cross-dependencies between packages
  - A solution to this could be the 'initial deploy' process we already have

### Checklist

- [ ] Tested changes end to end
- [x] Updated documentation
- [x] Requested one or more reviewers

### CODE REVIEW

**Comment by jhunter-twilio** [comment]
I like the simplicity of this for sure - @lvaladares, what do you think - does this address your needs?

**Review by lvaladares — APPROVED** [review]
yes @jhunter-twilio
cleanly an simply
lgtm

File: scripts/remove-features.mjs [hunk 1]
```diff
@@ -164,16 +156,16 @@ const performRemovals = async () => {
   
   if (!keepFeatures.includes("schedule-manager")) {
```
Comment (jhunter-twilio) [inline scripts/remove-features.mjs:157]: this area could get tricky to maintain in the future as we add in serverless packages but we can probably refactor this slightly later.

**Review by jhunter-twilio — APPROVED** [review]
this is looking great

**Comment by dremin** [comment]
Re-requesting review due to the docs changes.

**Review by jhunter-twilio — APPROVED** [review]
yeah i like these doc updates too - its starting to look pretty sensible now :) once we get the terraform changes in we can round it all off with documentation there too.

**Pull request merged**